### PR TITLE
Task8

### PR DESCRIPTION
**Pull request by dvoryak**

### CODE REVIEW

File: back-end/public/index.html [hunk 1]
```diff
@@ -0,0 +1,64 @@
+<html>
```
Comment (ScreamCat1) [inline back-end/public/index.html]: You don't need this file

File: back-end/public/stylesheets/style.css [hunk 2]
```diff
@@ -0,0 +1,12 @@
+img {
```
Comment (ScreamCat1) [inline back-end/public/stylesheets/style.css]: You don't need this file

File: back-end/services/core/authors/authors.middleware.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+module.exports = (server) => {	
+	router.get('/authors', (req, res, next) => {
+			let url_parts = url.parse(req.originalUrl, true),
+				query = url_parts.query,
+				queryStr = query.query,
+				authors = server.db.getState().authors;
+			
+			if (!!query.textFragment) {
+				authors = authors.filter((author) => author.name.concat(author.name).toUpperCase().indexOf(query.textFragment.toUpperCase()) >= 0);
```
Comment (ScreamCat1) [inline back-end/services/core/authors/authors.middleware.js]: Why do you need it, if you don't use this?

File: back-end/services/core/courses/courses.middleware.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+		let url_parts = url.parse(req.originalUrl, true);
+		const query = url_parts.query;
+		const from = query.start || 0;
+		let	to = from + parseInt(query.count, 10);
+		const sort = query.sort;
+		const id = query.id;
+		let	courses = server.db.getState().courses;
+
+		if (!!query.textFragment) {
+			courses = courses.filter((course) => course.name.concat(course.description).toUpperCase().indexOf(query.textFragment.toUpperCase()) >= 0);
```
Comment (ScreamCat1) [inline back-end/services/core/courses/courses.middleware.js]: The same

File: src/app/services/auth.service.ts [hunk 5]
```diff
@@ -1,34 +1,65 @@
...
+                    if (token) {
+                        this.setUser(token.token);
+                        this.$isAuthenticated.next(true);
+                        this.$currentLoggedUserSubject.next(token);
+                    }
+                    this.$isAuthenticated.next(false);
+                }),
+            ).subscribe();
+
+        return this.$isAuthenticated.asObservable();
```
Comment (ScreamCat1) [inline src/app/services/auth.service.ts]: You should not return `this.$isAuthenticated.asObservable();` you can create a variable for this and do something like you do with `this.$currentLoggedUserSubject.next(token);`

File: src/app/services/auth.service.ts [hunk 6]
```diff
@@ -1,34 +1,65 @@
...
+        localStorage.removeItem('token');
+        this.$currentLoggedUserSubject.next(null);
+    }
+
+    public getUser(): Observable<User> {
+        const tokenUser = this.getCurrentUser();
+        this.http.post<User>(`${SERVER_ENDPOINT}/${USER_INFO_PATH}`, {token: tokenUser})
+            .pipe(tap(user => this.$currentLoggedUserSubject.next(user)))
+            .subscribe();
+        return this.$currentLoggedUserSubject.asObservable();
```
Comment (ScreamCat1) [inline src/app/services/auth.service.ts]: When someone will subscribe to this pipe, `POST` method will be triggered, it is not a good way

File: src/app/services/auth.service.ts [hunk 7]
```diff
@@ -1,34 +1,65 @@
...
+    }
+
+    public isAuthenticated(): boolean {
+        return this.$isAuthenticated.value;
```
Comment (ScreamCat1) [inline src/app/services/auth.service.ts]: You can return Observable here

File: src/app/services/auth.service.ts [hunk 8]
```diff
@@ -1,34 +1,65 @@
...
+    }
+
+
+    private getCurrentUser(): string {
```
Comment (ScreamCat1) [inline src/app/services/auth.service.ts]: This method name is not equal that he does

File: src/app/services/auth.service.ts [hunk 9]
```diff
@@ -1,34 +1,65 @@
...
+        return localStorage.getItem('token');
+    }
+
+    private setUser(user: string): void {
+        localStorage.setItem('token', user);
```
Comment (ScreamCat1) [inline src/app/services/auth.service.ts]: The same as previous

File: src/app/core/guards/authorization.guard.ts [hunk 10]
```diff
@@ -6,15 +6,15 @@ import {AuthService} from '../../services/auth.service';
...
+    constructor(
       private router: Router,
       private authorizationService: AuthService
   ) {
   }
 
   canActivate(route: ActivatedRouteSnapshot, state: RouterStateSnapshot): boolean {
-    const currentUser = this.authorizationService.getUserInfo();
-    return !!currentUser;
+      return this.authorizationService.isAuthenticated();
```
Comment (ScreamCat1) [inline src/app/core/guards/authorization.guard.ts]: Guard can work with Observable, so you don't need to get value from Subject

File: back-end/services/core/courses/courses.routes.js [hunk 11]
```diff
@@ -0,0 +1,11 @@
+// REWRITE EXAMPLE
+const express = require('express');
+const jsonServer = require('json-server');
+const router = express.Router();
+
+router.use(jsonServer.rewriter({
+	'/courses': '/courses',
```
Comment (ScreamCat1) [inline back-end/services/core/courses/courses.routes.js]: You don't use it.
Please implement `courses fetching`

File: back-end/services/core/authors/authors.middleware.js (hunk 3 above)
Comment (dvoryak) [reply back-end/services/core/authors/authors.middleware.js]: I didn't code this. I have copied this code, it was provided by someone from skype conversation. Maybe I will use this functionality in future tasks.

File: src/app/services/auth.service.ts (hunk 6 above)
Comment (dvoryak) [reply src/app/services/auth.service.ts]: How can I refactor it?

File: back-end/services/core/authors/authors.middleware.js (hunk 3 above)
Comment (ScreamCat1) [reply back-end/services/core/authors/authors.middleware.js]: No, this functionality should be created in this task. Please see task

If you don't create it, please remove and that is it.

File: back-end/services/core/courses/courses.routes.js (hunk 11 above)
Comment (dvoryak) [reply back-end/services/core/courses/courses.routes.js]: done

**Pull request merged**